src/render.js

```diff
@@ -31,15 +29,19 @@ const resetGlobals = function() {
 
 resetGlobals();
 
-const runInContext = function(context, fn) {
-    return vm.runInContext("(" + fn.toString() + ")()", context);
-};
-
 /**
  * This method is executed whenever a render is needed. It is executed inside
  * the vm context.
  */
 const performRender = async () => {
+    if (window["__DEBUG_RENDER__"]) {
+        // To activate, set __DEBUG_RENDER__ to truthy on the vm context.
+        // Special debug entrypoint because this is run inside the vm context
+        // by turning it into a string, which means regular breakpoints don't
+        // work.
+        // eslint-disable-next-line no-debugger
+        debugger;
+    }
```

Comment:
The comment says it all. When debugging this method, it's easy to forget that we can't just add a regular breakpoint because of how it gets executed so, I added this hack.

Reply:
I think I understand this now.  It's so that you don't hit the debugger statement on every request.

Reply:
This is adding a `debugger` statement inside the VM context. We run this function inside JSDOM and as such, the function is converted to text so we can't use a regular editor breakpoint as this specific version of the code isn't what gets executed and our custom function to run this in the JSDOM context doesn't provide a sourcemap. So, we add this little conditional helper that says "if someone set the __DEBUG_RENDER__ flag to truthy, then act as though there's a breakpoint here". That way, folks can enable debugging this or not while the server is running, rather than having to edit the code and restart the server.